csf_tz/custom_api.py

```diff
@@ -2639,63 +2649,73 @@ def make_salary_components_and_structure(abbr ):
+            salary_component_doc.type = salary_component.get("type")
+            salary_component_doc.abbr = salary_component.get("abbr")
+            salary_component_doc.remove_if_zero_valued = salary_component.get(
+                "remove_if_zero_valued"
+            )
+            salary_component_doc.do_not_include_in_total = salary_component.get(
+                "do_not_include_in_total"
+            )
+            salary_component_doc.append(
+                "accounts", {"account": salary_component.get("account")}
+            )
+            salary_component_doc.insert()
+        else:
+            continue
+
+    salary_structure_doc_name = "Tanzania Mainland"
+    existing_salary_strusture = frappe.db.exists("Salary Structure",{"name": salary_structure_doc_name})
+    
+    if not existing_salary_strusture:
+        salary_structure_doc = frappe.new_doc("Salary Structure")
+        salary_structure_doc.name = salary_structure_doc_name
+        salary_structure_doc.is_active = "Yes"
+        for salary_components in salary_components_earnings_list:
+            salary_structure_doc.append(
+                "earnings",
+                {
+                    "salary_component": salary_components.get("salary_component"),
+                    "depends_on_payment_days": salary_components.get(
+                        "depends_on_payment_days"
+                    ),
+                    "is_tax_applicable": salary_components.get("is_tax_applicable"),
+                    "amount_based_on_formula": salary_components.get(
+                        "amount_based_on_formula"
+                    ),
+                    "do_not_include_in_total": salary_components.get(
+                        "do_not_include_in_total"
+                    ),
+                    "formula": salary_components.get("formula"),
+                },
+            )
+        for salary_components in salary_components_deduction_list:
+            salary_structure_doc.append(
+                "deductions",
+                {
+                    "salary_component": salary_components.get("salary_component"),
+                    "depends_on_payment_days": salary_components.get(
+                        "depends_on_payment_days"
+                    ),
+                    "is_tax_applicable": salary_components.get("is_tax_applicable"),
+                    "amount_based_on_formula": salary_components.get(
+                        "amount_based_on_formula"
+                    ),
+                    "do_not_include_in_total": salary_components.get(
+                        "do_not_include_in_total"
+                    ),
+                    "formula": salary_components.get("formula"),
+                },
+            )
+        salary_structure_doc.insert(ignore_permissions=True)
+        salary_structure_doc.submit()
+    else:
+        return "Salary Components and Structure are created successfully."
```

Comment:
This message should not fire only in the else clause, maybe keep a separate message in else clause which says ALREADY created.